---
title: "MR-SimSS: The algorithm"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{MR-SimSS: The algorithm}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```



### Introduction

A typical MR analysis is concerned with **estimating the causal effect, denoted by $\beta$, of a modifiable exposure, $X$ on a health-related outcome, $Y$**. When variables which confound the exposure-outcome relationship have not been observed, i.e. unmeasured confounders represented by $U$ are present, an MR study uses $n$ genetic variants, $G_1, G_2, \dots, G_n$, in an attempt to attain an unbiased causal effect estimate. 

The image below displays a causal diagram representing the <u> assumed relationship between genetic variant $G_j$, exposure $X$ and outcome $Y$ in an MR analysis</u>. $U$ represents unobserved confounders, while $\gamma_j$ is the effect of the variant on the exposure and $\beta$ is the true exposure-outcome causal effect.  For a particular individual, $G_j$ typically takes values 0, 1, or 2, $G_j \in \{0,1,2\}$,
depending on the minor allele count or genotype of the individual at genetic variant $j$.


<br>

<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/causal_diagram.jpg?raw=true"> 
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/causal_diagram.jpg?raw=true" width="90%">
</a>
</p>

<br>



[$\star$]{style="color: darkblue;"} **Note:** A table is included at the bottom of the page ([here](https://amandaforde.github.io/mr.simss/articles/MR-SimSS-algorithm.html#notation)) detailing all notation used in this vignette. 


Here, the focus is on **summary-level MR**, a form of MR in which the analysis is conducted with summary statistics that have been generated using two large non-overlapping, or partially overlapping, exposure and outcome data sets. These summary statistics typically take the form of variant-exposure association estimates, variant-outcome association estimates and their corresponding standard errors, i.e. $\left( \widehat{\gamma_j}, \widehat{\sigma_{X_j}}\right)$ and $\left( \widehat{\Gamma_j}, \widehat{\sigma_{Y_j}}\right)$ for each genetic variant $j = 1, \dots, n$. To be more specific, the exposure data is used to regress $X$ on each $G_j$, $j = 1, \dots, n$, independently, yielding regression coefficients, $\widehat{\gamma_j}$ and standard errors, $\widehat{\sigma_{X_j}}$ in the process. In a similar manner, the regression coefficients, $\widehat{\Gamma_j}$  and standard errors,  $\widehat{\sigma_{Y_j}}$  are generated by regressing $Y$ on each $G_j$ separately in the outcome data set.


[$\star$]{style="color: darkblue;"} <span style="color:darkgreen;">**Potential sample overlap is permitted**</span> between the two data sets and therefore, it is <u>not assumed that $\widehat{\gamma_j}$ and $\widehat{\Gamma_j}$ are independent</u> for each genetic variant $j$, i.e. it is possible that $\text{cov}\left( \widehat{\gamma_j}, \widehat{\Gamma_j} \right) \neq 0$.


### Intuition for MR-SimSS framework


The intuition for MR-SimSS stems from a solution to *Winner’s Curse* that could be implemented when individual-level data is available. 


#### <span style="color:darkblue;"> Individual-level: </span>


In this setting, we first suppose that we have individual-level data, i.e. genotype information, exposure measurements and/or outcome measurements, for a total of $N$ participants of the form $\{\textbf{G}, \textbf{X}, \textbf{Y}\}$. Here, $\textbf{G}$ represents the matrix of genotype information for each genetic variant and participant, i.e. $\textbf{G} = \{ \textbf{G}_1, \dots, \textbf{G}_{N} \} \in \mathbb{R}^{N \times n}$, where $\textbf{G}_{j} = \left[ G_{{1_j}}, \dots,G_{N_j}\right]^T$ contains the genotype information for the $j$th variant, and $\textbf{X} = \left[ X_1, \dots, X_{N} \right]^T$ is the vector of measured exposures. Similarly, $\textbf{Y} = \left[ Y_1, \dots, Y_{N} \right]^T$ is the vector of outcome values. It is possible that the vectors $\mathbf{X}$ and $\mathbf{Y}$ may have several missing values, as there may only be exposure or outcome values, but not both, available for certain individuals. We let $N_\text{overlap}$ denote the number of overlapping individuals, those who have measured values for both outcome and exposure. The proportion of sample overlap, $\frac{N_\text{overlap}}{N}$, can range from 0 to 1, i.e. no overlap to full overlap. 




[$\star$]{style="color: darkblue;"} **Eliminating** ***Winner's Curse***:

If this full data set $\{\mathbf{G},\mathbf{X},\mathbf{Y}\}$ were available, in order to avoid *Winner’s Curse*, we could first **randomly split it into two subsets**, with approximately $\pi N$ and $(1−\pi)N$ individuals in each set, respectively. Following this splitting, an exposure GWAS could be performed using the individual-level data in the first fraction or subset with $\pi N$ individuals, $\{\mathbf{G}_\pi,\mathbf{X}_\pi,\mathbf{Y}_\pi \}$, yielding <u> variant-exposure association estimates, $\widehat{\gamma_{\pi_j}}$for each variant</u>  $j = 1, \dots,n$. These estimated associations could then be used for the <u>sole purpose of selecting genetic instruments</u>. This would ensure that *Winner’s Curse* is avoided as the other independent subset, $\{\mathbf{G}_{1-\pi},\mathbf{X}_{1-\pi},\mathbf{Y}_{1-\pi}\}$, would be used to 
perform an exposure and outcome GWAS, generating variant-exposure, $\widehat{\gamma_{{1-\pi}_j}}$, and variant-outcome association estimates, $\widehat{\Gamma_{{1-\pi}_j}}$, for the MR study. This process is depicted in the figure below. 


<br>

<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_1.jpg?raw=true"> 
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_1.jpg?raw=true" width="90%">
</a>
</p>

<br>


[$\star$]{style="color: darkblue;"} **Weak instrument bias**:

Recall from the first vignette, ["Winner's Curse and weak instrument bias in MR"](https://amandaforde.github.io/mr.simss/articles/wc_wib.html), that if the variant-exposure and variant-outcome association estimates used in the MR analysis are derived from a single sample or two largely overlapping samples and if several weak instruments are incorporated, then bias towards the confounded observational association will be introduced into the MR causal effect estimate. When individual-level data is available, a potential solution to this as well as *Winner's Curse*, would be to consider **splitting the $1- \pi$ fraction into two fractions, $p$ and $1-p$**. As illustrated in the image below, the dataset $\{\mathbf{G}_{(1-\pi)(p)},\mathbf{X}_{(1-\pi)(p)},\mathbf{Y}_{(1-\pi)(p)}\}$ could be used to perform an exposure GWAS and supply variant-exposure association estimates, $\widehat{\gamma_{{(1-\pi)(p)}_j}}$, while an outcome GWAS could be conducted with $\{\mathbf{G}_{(1-\pi)(1-p)},\mathbf{X}_{(1-\pi)(1-p)},\mathbf{Y}_{(1-\pi)(1-p)}\}$, giving variant-outcome association estimates, $\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}$. A suitable two-sample summary-level MR method could then be applied to the resulting summary statistics: $\{ \widehat{\gamma_{{(1-\pi)(p)}_j}}, \widehat{\Gamma_{{(1-\pi)(1-p)}_j}} \}$. 




<br>

<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_2.jpg?raw=true"> 
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_2.jpg?raw=true" width="100%">
</a>
</p>

<br>



Unfortunately, this splitting process, such as the one illustrated above, results in a <u>great loss in sample size, and hence power</u>. However, a reduction in the variability of the resulting MR causal effect estimate can be acquired by **repeating the random splitting and performing GWASs in each subset many, many times**, and then, averaging over the causal effect estimates obtained on each iteration. 

However, this is **not a very feasible solution**, as in addition to <u>requiring individual-level data</u>, a major limitation of this approach is its <u>severe computational demands</u>, in terms of both run time and memory usage.

<br>

[$\star$]{style="color: darkblue;"} **MR-SimSS seeks to imitate this repeated sample splitting process, using only GWAS summary statistics.**



### How does MR-SimSS work?

Firstly, MR-SimSS works with summary-level data. Thus, it assumes that an exposure and outcome GWAS have been performed, providing variant-exposure and variant-outcome association estimates, $\{\widehat{\gamma_{j}}, \widehat{\Gamma_j}\}$, in which there may be a number of overlapping individuals between the two GWASs, $N_\text{overlap}$. As MR-SimSS has been constructed with the intention to imitate the above repeated sample splitting, it first considers splitting the dataset into two fractions, $\pi$ and $1-\pi$, and simulates values for the association estimates in the first fraction, $\{\widehat{\gamma_{{\pi}_j}}, \widehat{\Gamma_{{\pi}_j}}\}$, conditional on those in the full dataset, $\{\widehat{\gamma_{j}}, \widehat{\Gamma_j}\}$. Thus, the first step of MR-SimSS involves simulating values for $\{\widehat{\gamma_{{\pi}_j}}, \widehat{\Gamma_{{\pi}_j}}\}$, using the asymptotic conditional distribution of .... For genetic variant $j$, the following formula for the asymptotic conditional distribution has been derived: 



Here, $\rho$ represents the correlation between the exposure and the outcome, $\rho = \text{cor}(X,Y)$, which may differ from $\beta$, the true exposure-outcome causal effect, due to confounding. See the vignette ['Deriving
MR-SimSS'](https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html) for a detailed outline of the derivation of this asymptotic distribution. 

The corresponding estimated standard errors in the first fraction, $\{ \widehat{\sigma_{X_{\pi,j}}},\widehat{\sigma_{Y_{\pi,j}}}\}$ can be obtained by approximating then with $\sigma_{X_{\pi,j}} \sim \sqrt{\frac{1}{\pi}} \cdot \widehat{\sigma_{X_{j}}}$ and $\sigma_{Y_{\pi,j}} \sim \sqrt{\frac{1}{\pi}} \cdot \widehat{\sigma_{Y_{j}}}$, respectively. 


The genetic instruments are then selected according to their z-statistics, $z_{X_{\pi,j}} = \frac{\widehat{\gamma_{\pi_j}}}{\widehat{\sigma_{X_{\pi,j}}}}, j=1, \dots, n$ in this fraction, i.e. genetic variants are chosen to be incorporated into the  MR analysis which have corresponding p-values less than the genome-wide significance threshold of $5 \times 10^{-8}$. For each genetic variant, the association estimates that would have been obtained if GWASs were conducted in the second fraction, $1−\pi$, can be generated using $\widehat{\gamma_j} \approx \pi\widehat{\gamma_{\pi_j}} + (1-\pi)\widehat{\gamma_{{(1-\pi)}_j}}$ and $\widehat{\Gamma_j} \approx \pi\widehat{\Gamma_{\pi_j}} + (1-\pi)\widehat{\Gamma_{{(1-\pi)}_j}}$. Note that these relationships hold due to the asymptotic linearity of maximum likelihood estimates but therefore, are only approximate to order 1/N. The estimated variant-exposure and variant-outcome associations and corresponding standard errors in the $1-\pi$ fraction are obtained as:


We first considered a 2-split approach in which the estimated associations and standard errors from the above equation were then inputted into a summary-level MR method of choice, such as the IVW method.


<br>

<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_1.jpg?raw=true"> 
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_1.jpg?raw=true" width="100%">
</a>
</p>

<br>


However, we recognised that this could induce a notable amount of weak instrument bias in the direction of the confounded observational association between the exposure and the outcome if there existed a large degree of overlap between the sample sets used in the exposure and outcome GWASs. Therefore, we also propose a 3-split approach which in addition to eliminating Winner’s Curse bias, can only result in weak instrument bias towards the null, irrespective of the proportion of sample overlap. With the 3-split approach, we imagine as if we go on to split the individual-level data in the $1-\pi$ fraction into two fractions, $p$ and $1− p$, and perform an exposure GWAS in the $p$ fraction and an outcome GWAS in the $1− p$ fraction.

Our method imitates this idea by simulating values for the estimated variant-exposure and variant-outcome associations, $\{\widehat{\gamma_{{(1-\pi)p}_j}}, \widehat{\Gamma_{{(1-\pi)p}_j}}\}$, respectively, conditional on the generated association estimates in the $1-\pi$ fraction, $\{\widehat{\gamma_{{(1-\pi)}_j}}, \widehat{\Gamma_{{(1-\pi)}_j}}\}$. The asymptotic conditional distribution of .... takes a similar format to and an expression for it can be found in the vignette ['Deriving
MR-SimSS'](https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html). 



Using the asymptotic linearity of maximum likelihood estimates, $\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}$ is obtained using $\widehat{\Gamma_{{(1-\pi)(1-p)}_j}} \approx \frac{\widehat{\Gamma_{{(1-\pi)}_j}}-p\widehat{\Gamma_{{(1-\pi)p}_j}}}{1-p}$. In addition, the estimated standard errors, $\widehat{\sigma_{X_{(1-\pi)p,j}}}$ and  $\widehat{\sigma_{X_{(1-\pi)(1-p),j}}}$ can be approximated by ..., respectively.


Therefore, the values  for each variant $j=1, \dots, n_{\text{sig}}$ in which all genetic variants $j=1, \dots, n_{\text{sig}}$ have been selected in the first $\pi$ fraction, i.e. satisfy the condition..., are
inputted into the MR method of choice. 








<br>

<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_2.jpg?raw=true"> 
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_2.jpg?raw=true" width="100%">
</a>
</p>

<br>



In order to reduce the variability in the final MR-SimSS causal effect estimate and ensure its stability, we suggest repeating this simulated sample splitting process many times, at least $N_\text{iter} = 1,000$ iterations.

 

The final overall estimate for $\hat\beta$, the exposure-outcome causal effect, is then equivalent to the average of the estimates generated by each iteration of the process.


An estimate of the causal effect obtained at iteration $k$ of the simulated sample splitting process can be denoted by $\hat\beta^{(k)}$. For example, we could have $\hat\beta^{(k)} = \hat\beta^{(k)}_{\text{IVW}}$ or $\hat\beta^{(k)} = \hat\beta^{(k)}_{\text{RAPS}}$, dependent on which summary-level MR method is employed. The final estimate for the causal effect supplied by MR-SimSS is then denoted by $\overline{\hat\beta} = \frac{1}{N_{\text{iter}}} \sum_{k=1}^{N_{\text{iter}}} \hat\beta^{(k)}$, the average
of the estimates obtained at each iteration.

A suitable value for $\text{se}(\overline{\hat\beta})$, the standard error of the MR-SimSS causal effect estimate, can be obtained using the following expression:

$$\text{se}(\overline{\hat\beta}) = \sqrt{\frac{\sum_{k=1}^{N_{\text{iter}}} \left[ \left( \text{se}\left( \hat\beta^{(k)} \right)\right)^2 \right] - \sum_{k=1}^{N_{\text{iter}}} \left[ \hat\beta^{(k)} - \overline{\hat\beta}\right]^2}{N_\text{iter}}}$$ 
Note that $\text{se}\left(\hat\beta^{(k)}\right)$ is the standard error of the causal effect estimate at iteration $k$, which is typically provided by the summary-level MR method of choice. 

<br>

[$\star$]{style="color: darkblue;"} **Note:** The asymptotic conditional distribution equation suggests that in order to implement MR-SimSS, if the number of overlapping samples involved in the exposure and outcome GWASs, $N_\text{overlap}$, is known to differ from zero, $N_\text{overlap} \neq 0$, then this number $N_\text{overlap}$ as well as the correlation between the exposure and outcome, $\rho$, must be provided. In general, $N_\text{overlap}$ may be known, especially if the GWASs have been performed on a single fully overlapping sample, and an estimate for $\rho$ could be obtained from the literature. However, if neither $N_\text{overlap}$ nor an estimate for $\rho$ can be supplied, an alternative strategy is provided [here]() which seeks to estimate a value for $\lambda = \frac{N_\text{overlap}\rho}{\sqrt{N_X N_Y}}$, using only the available **summary-level data of the genome-wide set of genetic variants** i.e. $\{ \widehat{\gamma_j}, \widehat{\sigma_{X_j}}, \widehat{\Gamma_j}, \widehat{\sigma_{Y_j}}\}$. 

<br>

[$\star$]{style="color: darkblue;"} **Note:** It is mentioned that several iterations of our approach, ideally $N_{\text{iter}} \approx 1000$ iterations, is required in order to ensure a stable causal effect estimate. Unfortunately, repeatedly simulating values for $\{\gamma_{\pi_j}, \Gamma_{\pi_j}\}$ and $\{\gamma_{{(1-\pi)p}_j}, \Gamma_{{(1-\pi)p}_j}\}$ for SNPs $j = 1, \dots ,N$ several times can make MR-SimSS a computationally intensive process, especially in comparison with other summary-level MR methods. Recognising that irrespective of the genetic architecture of the exposure, there will likely be many variants which will never be selected across the $N_\text{iter}$ iterations, we therefore establish a subset of genetic variants according to the cumulative sums of their probabilities, $f\left( z_{X_{\pi,j}} > 5.45\right)$. Further details regarding can be found here.




### Assumptions made by MR-SimSS 

As with many statistical methods, the MR-SimSS framework makes <span style="color:darkblue;">**several assumptions**</span>:

<ul>
  <li> Firstly, the genetic association estimates with respect to both exposure and outcome are assumed to be **normally distributed** for each $j \in \{1, \dots, n\}$, i.e. $\widehat{\gamma_j} \sim N\left(\gamma_j, \sigma_{X_j}^2 \right)$ and $\widehat{\Gamma_j} \sim N\left(\Gamma_j, \sigma_{Y_j}^2 \right)$. This normality assumption frequently appears in the MR literature and is considered to be reasonable, based on asymptotic considerations, since <u>estimated associations are obtained from large GWASs</u>. 
  </li> 
  
  <li> It is assumed that the variance ratio $\frac{\sigma_{X_j}^2}{\sigma_{Y_j}^2}$for $j=1, \dots, n$ is both bounded away from zero and infinity. 
  </li>
  
  <li> The set of random variables **$\{\widehat{\gamma_j}, j=1, \dots, n\}$ are assumed to be mutually independent**, while similarly, **$\{\widehat{\Gamma_j}, j=1, \dots, n \}$ are assumed to be mutually independent**. However, $\gamma_j$ and $\Gamma_j$ are not necessarily independent due to participant overlap between the exposure and outcome data sets. To satisfy this assumption regarding independence among variants, a common step in MR studies is to first <u>employ LD pruning or clumping to produce a large set of uncorrelated genetic variants</u>, which the $p$ genetic instruments are then selected from. 
  </li>
  
  <li> The assumption that the **standard deviations $\sigma_{X_j}^2$ and $\sigma_{Y_j}^2$ for $j \in \{1, \dots, n\}$ are known**, is also made here. In fact, most, if not all, MR analyses conducted today assume that $\widehat{\sigma_{X_j}} \approx \sigma_{X_j}$ and $\widehat{\sigma_{Y_j}} \approx \sigma_{Y_j}$. This practice is deemed appropriate by many as large GWAS sample sizes ensure that the <u>variances of the genetic association estimates are computed with a great degree of accuracy</u>.
  
  </li>
  
</ul>












### Notation 



<br>

<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_MR/blob/main/table.jpg?raw=true"> 
<img src="https://github.com/amandaforde/winnerscurse_MR/blob/main/table.jpg?raw=true" width="100%">
</a>
</p>

<br>
