---
title: "Performing MR-SimSS"
output: 
  rmarkdown::html_vignette:
    toc: yes
vignette: >
  %\VignetteIndexEntry{Performing MR-SimSS}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup}
library(mr.simss)
```


This vignette .. brief introduction .... 



What do I want to include in here? Demonstrate example how to use mr_simss and also est_lambda 

First create a toy data set...

### Creating a toy data set

Establish function to create toy data set.. specify certain parameters etc.... 

### Using `mr_simss`

After toy data set is made, we can now consider applying method ... detailed discussion of how to do that and the various options included ... 


### Using `est_lambda`

As well as estimates of the variant-exposure and variant-outcome associations and corresponding standard errors, the method also requires *(a)* knowledge of the number of samples in both outcome and exposure GWASs, the number of overlapping samples between the two GWASs, and the correlation between the exposure and the outcome, or *(b)* an estimate of the correlation between the SNP-outcome and SNP-exposure effect sizes. Due to this, this R package also contains a ***supplementary function***, `est_lambda`, which allows users to obtain an *unbiased* estimate for *lambda*, the correlation between the variant-outcome and variant-exposure effect sizes, using a conditional log-likelihood approach. It is recommended to use this function when the fraction of overlap and the correlation between exposure and outcome are unknown. Merely the GWAS **summary statistics** of a large set of *unpruned* variants or single nucleotide polymorphisms (SNPs) are required in order to obtain an estimate for *lambda*.


Important to have a section on Runtime here... 

### Runtime
