<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MR-SimSS: Mendelian Randomisation (MR) method that combats Winner's Curse using a simulated sample splitting approach  • mr.simss</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/lumen/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="MR-SimSS: Mendelian Randomisation (MR) method that combats Winner's Curse using a simulated sample splitting approach ">
<meta property="og:description" content="Designed to provide users with a method, namely MR-SimSS, which uses simulated sample splitting in order to alleviate 
 Winner's Curse bias in MR causal effect estimates. This approach also takes into account sample overlap between the exposure and outcome genome-wide association studies. It 
 uses summary statistics from genome-wide association studies and works in combination with existing MR methods, such as IVW and MR-RAPS.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">mr.simss</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/index.html">Articles</a>
</li>
<li>
  <a href="reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/amandaforde/mr.simss" class="external-link">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="simulated-sample-splitting-to-overcome-winners-curse-in-summary-level-mendelian-randomization">Simulated sample splitting to overcome <em>Winner’s Curse</em> in summary-level Mendelian Randomization<a class="anchor" aria-label="anchor" href="#simulated-sample-splitting-to-overcome-winners-curse-in-summary-level-mendelian-randomization"></a>
</h1></div>
<p><br></p>
<p><span style="color: darkred;"><span class="math inline">⋆</span></span> <span style="color:darkred;"><strong>Note:</strong> </span> <strong><code>mr.simss</code> is still under <u> active development</u>.</strong></p>
<p><br></p>
<p>This R package, <code>mr.simss</code>, has been designed to allow users to easily implement the Mendelian Randomization (MR) framework known as <strong>MR Simulated Sample Splitting (MR-SimSS)</strong>. The MR-SimSS paradigm aims to <u>eliminate bias induced by <em>Winner’s Curse</em></u> in MR causal effect estimates, using only <strong>summary statistics</strong> from genome-wide association studies (GWASs). The approach is based on a <span style="color:darkblue;"><strong>repeated simulated sample splitting</strong></span> procedure and works in combination with existing summary-level MR methods, such as IVW and MR-RAPS. In addition, the framework is designed to <u>assist in circumventing biases (e.g. weak instrument bias) arising due to sample overlap</u> between exposure and outcome GWASs, and thus, can be applied in both one-sample and two-sample cases. The <u>main function</u> in this package for executing MR-SimSS is <code>mr_simss</code>. <code>mr_simss</code> has several parameters which users can adjust based on their desired form of method implementation. Further discussion regarding these parameters can be viewed in the first vignette, <a href="https://amandaforde.github.io/mr.simss/articles/perform-MR-SimSS.html" class="external-link">‘Performing MR-SimSS’</a>, while a detailed derivation of the MR-SimSS framework can be found in the second vignette, <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html" class="external-link">‘Performing MR-SimSS’</a>.</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>You can install the current version of <code>mr.simss</code> from <strong>GitHub</strong> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"amandaforde/mr.simss"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/amandaforde/mr.simss" class="external-link">mr.simss</a></span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="mr-simss-overview">MR-SimSS Overview<a class="anchor" aria-label="anchor" href="#mr-simss-overview"></a>
</h3>
<p><em>small paragraph here providing brief description of method.. building on what was already stated in the introduction</em></p>
<p>The method in question, namely MR Simulated Sample Splitting (MR-SimSS), aims to extend the benefits of the ‘threesample’ design approach to the single sample setting, ensuring minimal weak instrument bias and avoiding Winner’s Curse in the process. In our work, we have explicitly focused on establishing an MR method that can be used when only GWAS summary-level data, in the form of variant-exposure and variant-outcome association estimates and standard errors, is available. It is assumed that this data is derived from two large exposure and outcome GWASs, which have been conducted with two sets of non-overlapping, partially overlapping or fully overlapping samples. Being a summary-level approach, the proposed method benefits from a key advantage, in the sense that it is very computationally efficient. MR-SimSS works by imitating the splitting of an individual-level data set into three portions, with the first fraction reserved for instrument selection and the other two used to independently estimate the variant-exposure and variant-outcome associations. Taking advantage of asymptotic conditional distributions, MR-SimSS repetitively simulates association estimates in each fraction conditional on the known estimates in the full data set. On each iteration, instrument variants are selected according to the simulated estimates in the first fraction, while a two-sample MR method, such as IVW or MR-RAPS, is then applied to the estimates from the other two fractions to obtain a causal effect estimate. MR-SimSS then averages over the MR estimates generated on each iteration to provide a final exposure-outcome causal effect estimate. This act of averaging over a great number of iterations ensures reduced variance in the final estimate.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/amandaforde/mr.simss/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/amandaforde/mr.simss/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing mr.simss</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Amanda Forde <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7940-3413" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Amanda Forde.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
