<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MR-SimSS: The algorithm • mr.simss</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/lumen/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MR-SimSS: The algorithm">
<meta property="og:description" content="mr.simss">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mr.simss</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/amandaforde/mr.simss" class="external-link">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MR-SimSS: The algorithm</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/amandaforde/mr.simss/blob/HEAD/vignettes/MR-SimSS-algorithm.Rmd" class="external-link"><code>vignettes/MR-SimSS-algorithm.Rmd</code></a></small>
      <div class="hidden name"><code>MR-SimSS-algorithm.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>A typical MR analysis is concerned with <strong>estimating the causal
effect, denoted by <span class="math inline">\(\beta\)</span>, of a
modifiable exposure, <span class="math inline">\(X\)</span> on a
health-related outcome, <span class="math inline">\(Y\)</span></strong>.
When variables which confound the exposure-outcome relationship have not
been observed, i.e. unmeasured confounders represented by <span class="math inline">\(U\)</span> are present, an MR study uses <span class="math inline">\(n\)</span> genetic variants, <span class="math inline">\(G_1, G_2, \dots, G_n\)</span>, in an attempt to
attain an unbiased causal effect estimate.</p>
<p>The image below displays a causal diagram representing the <u>
assumed relationship between genetic variant <span class="math inline">\(G_j\)</span>, exposure <span class="math inline">\(X\)</span> and outcome <span class="math inline">\(Y\)</span> in an MR analysis</u>. <span class="math inline">\(U\)</span> represents unobserved confounders,
while <span class="math inline">\(\gamma_j\)</span> is the effect of the
variant on the exposure and <span class="math inline">\(\beta\)</span>
is the true exposure-outcome causal effect. For a particular individual,
<span class="math inline">\(G_j\)</span> typically takes values 0, 1, or
2, <span class="math inline">\(G_j \in \{0,1,2\}\)</span>, depending on
the minor allele count or genotype of the individual at genetic variant
<span class="math inline">\(j\)</span>.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/causal_diagram.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/causal_diagram.jpg?raw=true" width="90%"></a>
</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Note:</strong> A
table is included at the bottom of the page (<a href="https://amandaforde.github.io/mr.simss/articles/MR-SimSS-algorithm.html#notation" class="external-link">here</a>)
detailing all notation used in this vignette.</p>
<p>Here, the focus is on <strong>summary-level MR</strong>, a form of MR
in which the analysis is conducted with summary statistics that have
been generated using two large non-overlapping, or partially
overlapping, exposure and outcome data sets. These summary statistics
typically take the form of variant-exposure association estimates,
variant-outcome association estimates and their corresponding standard
errors, i.e. <span class="math inline">\(\left( \widehat{\gamma_j},
\widehat{\sigma_{X_j}}\right)\)</span> and <span class="math inline">\(\left( \widehat{\Gamma_j},
\widehat{\sigma_{Y_j}}\right)\)</span> for each genetic variant <span class="math inline">\(j = 1, \dots, n\)</span>. To be more specific, the
exposure data is used to regress <span class="math inline">\(X\)</span>
on each <span class="math inline">\(G_j\)</span>, <span class="math inline">\(j = 1, \dots, n\)</span>, independently, yielding
regression coefficients, <span class="math inline">\(\widehat{\gamma_j}\)</span> and standard errors,
<span class="math inline">\(\widehat{\sigma_{X_j}}\)</span> in the
process. In a similar manner, the regression coefficients, <span class="math inline">\(\widehat{\Gamma_j}\)</span> and standard errors,
<span class="math inline">\(\widehat{\sigma_{Y_j}}\)</span> are
generated by regressing <span class="math inline">\(Y\)</span> on each
<span class="math inline">\(G_j\)</span> separately in the outcome data
set.</p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <span style="color:darkgreen;"><strong>Potential sample overlap is
permitted</strong></span> between the two data sets and therefore, it is
<u>not assumed that <span class="math inline">\(\widehat{\gamma_j}\)</span> and <span class="math inline">\(\widehat{\Gamma_j}\)</span> are independent</u>
for each genetic variant <span class="math inline">\(j\)</span>, i.e. it
is possible that <span class="math inline">\(\text{cov}\left(
\widehat{\gamma_j}, \widehat{\Gamma_j} \right) \neq 0\)</span>.</p>
</div>
<div class="section level3">
<h3 id="intuition-for-mr-simss-framework">Intuition for MR-SimSS framework<a class="anchor" aria-label="anchor" href="#intuition-for-mr-simss-framework"></a>
</h3>
<p>The intuition for MR-SimSS stems from a solution to <em>Winner’s
Curse</em> that could be implemented when individual-level data is
available.</p>
<div class="section level4">
<h4 id="individual-level">
<span style="color:darkblue;"> Individual-level: </span><a class="anchor" aria-label="anchor" href="#individual-level"></a>
</h4>
<p>In this setting, we first suppose that we have individual-level data,
i.e. genotype information, exposure measurements and/or outcome
measurements, for a total of <span class="math inline">\(N\)</span>
participants of the form <span class="math inline">\(\{\textbf{G},
\textbf{X}, \textbf{Y}\}\)</span>. Here, <span class="math inline">\(\textbf{G}\)</span> represents the matrix of
genotype information for each genetic variant and participant,
i.e. <span class="math inline">\(\textbf{G} = \{ \textbf{G}_1, \dots,
\textbf{G}_{N} \} \in \mathbb{R}^{N \times n}\)</span>, where <span class="math inline">\(\textbf{G}_{j} = \left[ G_{{1_j}},
\dots,G_{N_j}\right]^T\)</span> contains the genotype information for
the <span class="math inline">\(j\)</span>th variant, and <span class="math inline">\(\textbf{X} = \left[ X_1, \dots, X_{N}
\right]^T\)</span> is the vector of measured exposures. Similarly, <span class="math inline">\(\textbf{Y} = \left[ Y_1, \dots, Y_{N}
\right]^T\)</span> is the vector of outcome values. It is possible that
the vectors <span class="math inline">\(\mathbf{X}\)</span> and <span class="math inline">\(\mathbf{Y}\)</span> may have several missing
values, as there may only be exposure or outcome values, but not both,
available for certain individuals. We let <span class="math inline">\(N_\text{overlap}\)</span> denote the number of
overlapping individuals, those who have measured values for both outcome
and exposure. The proportion of sample overlap, <span class="math inline">\(\frac{N_\text{overlap}}{N}\)</span>, can range
from 0 to 1, i.e. no overlap to full overlap.</p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Eliminating</strong>
<strong><em>Winner’s Curse</em></strong>:</p>
<p>If this full data set <span class="math inline">\(\{\mathbf{G},\mathbf{X},\mathbf{Y}\}\)</span> were
available, in order to avoid <em>Winner’s Curse</em>, we could first
<strong>randomly split it into two subsets</strong>, with approximately
<span class="math inline">\(\pi N\)</span> and <span class="math inline">\((1−\pi)N\)</span> individuals in each set,
respectively. Following this splitting, an exposure GWAS could be
performed using the individual-level data in the first fraction or
subset with <span class="math inline">\(\pi N\)</span> individuals,
<span class="math inline">\(\{\mathbf{G}_\pi,\mathbf{X}_\pi,\mathbf{Y}_\pi
\}\)</span>, yielding <u> variant-exposure association estimates, <span class="math inline">\(\widehat{\gamma_{\pi_j}}\)</span>for each
variant</u> <span class="math inline">\(j = 1, \dots,n\)</span>. These
estimated associations could then be used for the <u>sole purpose of
selecting genetic instruments</u>. This would ensure that <em>Winner’s
Curse</em> is avoided as the other independent subset, <span class="math inline">\(\{\mathbf{G}_{1-\pi},\mathbf{X}_{1-\pi},\mathbf{Y}_{1-\pi}\}\)</span>,
would be used to perform an exposure and outcome GWAS, generating
variant-exposure, <span class="math inline">\(\widehat{\gamma_{{1-\pi}_j}}\)</span>, and
variant-outcome association estimates, <span class="math inline">\(\widehat{\Gamma_{{1-\pi}_j}}\)</span>, for the MR
study. This process is depicted in the figure below.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_1.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_1.jpg?raw=true" width="90%"></a>
</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Weak instrument
bias</strong>:</p>
<p>Recall from the first vignette, <a href="https://amandaforde.github.io/mr.simss/articles/wc_wib.html" class="external-link">“Winner’s
Curse and weak instrument bias in MR”</a>, that if the variant-exposure
and variant-outcome association estimates used in the MR analysis are
derived from a single sample or two largely overlapping samples and if
several weak instruments are incorporated, then bias towards the
confounded observational association will be introduced into the MR
causal effect estimate. When individual-level data is available, a
potential solution to this as well as <em>Winner’s Curse</em>, would be
to consider <strong>splitting the <span class="math inline">\(1-
\pi\)</span> fraction into two fractions, <span class="math inline">\(p\)</span> and <span class="math inline">\(1-p\)</span></strong>. As illustrated in the image
below, the dataset <span class="math inline">\(\{\mathbf{G}_{(1-\pi)(p)},\mathbf{X}_{(1-\pi)(p)},\mathbf{Y}_{(1-\pi)(p)}\}\)</span>
could be used to perform an exposure GWAS and supply variant-exposure
association estimates, <span class="math inline">\(\widehat{\gamma_{{(1-\pi)(p)}_j}}\)</span>, while
an outcome GWAS could be conducted with <span class="math inline">\(\{\mathbf{G}_{(1-\pi)(1-p)},\mathbf{X}_{(1-\pi)(1-p)},\mathbf{Y}_{(1-\pi)(1-p)}\}\)</span>,
giving variant-outcome association estimates, <span class="math inline">\(\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}\)</span>. A
suitable two-sample summary-level MR method could then be applied to the
resulting summary statistics: <span class="math inline">\(\{
\widehat{\gamma_{{(1-\pi)(p)}_j}}, \widehat{\Gamma_{{(1-\pi)(1-p)}_j}}
\}\)</span>.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_2.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_2.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
<p>Unfortunately, this splitting process, such as the one illustrated
above, results in a <u>great loss in sample size, and hence power</u>.
However, a reduction in the variability of the resulting MR causal
effect estimate can be acquired by <strong>repeating the random
splitting and performing GWASs in each subset many, many times</strong>,
and then, averaging over the causal effect estimates obtained on each
iteration.</p>
<p>However, this is <strong>not a very feasible solution</strong>, as in
addition to <u>requiring individual-level data</u>, a major limitation
of this approach is its <u>severe computational demands</u>, in terms of
both run time and memory usage.</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>MR-SimSS seeks to
imitate this repeated sample splitting process, using only GWAS summary
statistics.</strong></p>
</div>
</div>
<div class="section level3">
<h3 id="how-does-mr-simss-work">How does MR-SimSS work?<a class="anchor" aria-label="anchor" href="#how-does-mr-simss-work"></a>
</h3>
<p>Firstly, MR-SimSS works with summary-level data. Thus, it assumes
that an exposure and outcome GWAS have been performed, providing
variant-exposure and variant-outcome association estimates, <span class="math inline">\(\{\widehat{\gamma_{j}},
\widehat{\Gamma_j}\}\)</span>, in which there may be a number of
overlapping individuals between the two GWASs, <span class="math inline">\(N_\text{overlap}\)</span>. As MR-SimSS has been
constructed with the intention to imitate the above repeated sample
splitting, it first considers splitting the dataset into two fractions,
<span class="math inline">\(\pi\)</span> and <span class="math inline">\(1-\pi\)</span>, and simulates values for the
association estimates in the first fraction, <span class="math inline">\(\{\widehat{\gamma_{{\pi}_j}},
\widehat{\Gamma_{{\pi}_j}}\}\)</span>, conditional on those in the full
dataset, <span class="math inline">\(\{\widehat{\gamma_{j}},
\widehat{\Gamma_j}\}\)</span>. Thus, the first step of MR-SimSS involves
simulating values for <span class="math inline">\(\{\widehat{\gamma_{{\pi}_j}},
\widehat{\Gamma_{{\pi}_j}}\}\)</span>, using the asymptotic conditional
distribution of …. For genetic variant <span class="math inline">\(j\)</span>, the following formula for the
asymptotic conditional distribution has been derived:</p>
<p>Here, <span class="math inline">\(\rho\)</span> represents the
correlation between the exposure and the outcome, <span class="math inline">\(\rho = \text{cor}(X,Y)\)</span>, which may differ
from <span class="math inline">\(\beta\)</span>, the true
exposure-outcome causal effect, due to confounding. See the vignette <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html" class="external-link">‘Deriving
MR-SimSS’</a> for a detailed outline of the derivation of this
asymptotic distribution.</p>
<p>The corresponding estimated standard errors in the first fraction,
<span class="math inline">\(\{
\widehat{\sigma_{X_{\pi,j}}},\widehat{\sigma_{Y_{\pi,j}}}\}\)</span> can
be obtained by approximating then with <span class="math inline">\(\sigma_{X_{\pi,j}} \sim \sqrt{\frac{1}{\pi}} \cdot
\widehat{\sigma_{X_{j}}}\)</span> and <span class="math inline">\(\sigma_{Y_{\pi,j}} \sim \sqrt{\frac{1}{\pi}} \cdot
\widehat{\sigma_{Y_{j}}}\)</span>, respectively.</p>
<p>The genetic instruments are then selected according to their
z-statistics, <span class="math inline">\(z_{X_{\pi,j}} =
\frac{\widehat{\gamma_{\pi_j}}}{\widehat{\sigma_{X_{\pi,j}}}}, j=1,
\dots, n\)</span> in this fraction, i.e. genetic variants are chosen to
be incorporated into the MR analysis which have corresponding p-values
less than the genome-wide significance threshold of <span class="math inline">\(5 \times 10^{-8}\)</span>. For each genetic
variant, the association estimates that would have been obtained if
GWASs were conducted in the second fraction, <span class="math inline">\(1−\pi\)</span>, can be generated using <span class="math inline">\(\widehat{\gamma_j} \approx
\pi\widehat{\gamma_{\pi_j}} +
(1-\pi)\widehat{\gamma_{{(1-\pi)}_j}}\)</span> and <span class="math inline">\(\widehat{\Gamma_j} \approx
\pi\widehat{\Gamma_{\pi_j}} +
(1-\pi)\widehat{\Gamma_{{(1-\pi)}_j}}\)</span>. Note that these
relationships hold due to the asymptotic linearity of maximum likelihood
estimates but therefore, are only approximate to order 1/N. The
estimated variant-exposure and variant-outcome associations and
corresponding standard errors in the <span class="math inline">\(1-\pi\)</span> fraction are obtained as:</p>
<p>We first considered a 2-split approach in which the estimated
associations and standard errors from the above equation were then
inputted into a summary-level MR method of choice, such as the IVW
method.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_1.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_1.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
<p>However, we recognised that this could induce a notable amount of
weak instrument bias in the direction of the confounded observational
association between the exposure and the outcome if there existed a
large degree of overlap between the sample sets used in the exposure and
outcome GWASs. Therefore, we also propose a 3-split approach which in
addition to eliminating Winner’s Curse bias, can only result in weak
instrument bias towards the null, irrespective of the proportion of
sample overlap. With the 3-split approach, we imagine as if we go on to
split the individual-level data in the <span class="math inline">\(1-\pi\)</span> fraction into two fractions, <span class="math inline">\(p\)</span> and <span class="math inline">\(1−
p\)</span>, and perform an exposure GWAS in the <span class="math inline">\(p\)</span> fraction and an outcome GWAS in the
<span class="math inline">\(1− p\)</span> fraction.</p>
<p>Our method imitates this idea by simulating values for the estimated
variant-exposure and variant-outcome associations, <span class="math inline">\(\{\widehat{\gamma_{{(1-\pi)p}_j}},
\widehat{\Gamma_{{(1-\pi)p}_j}}\}\)</span>, respectively, conditional on
the generated association estimates in the <span class="math inline">\(1-\pi\)</span> fraction, <span class="math inline">\(\{\widehat{\gamma_{{(1-\pi)}_j}},
\widehat{\Gamma_{{(1-\pi)}_j}}\}\)</span>. The asymptotic conditional
distribution of …. takes a similar format to and an expression for it
can be found in the vignette <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html" class="external-link">‘Deriving
MR-SimSS’</a>.</p>
<p>Using the asymptotic linearity of maximum likelihood estimates, <span class="math inline">\(\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}\)</span> is
obtained using <span class="math inline">\(\widehat{\Gamma_{{(1-\pi)(1-p)}_j}} \approx
\frac{\widehat{\Gamma_{{(1-\pi)}_j}}-p\widehat{\Gamma_{{(1-\pi)p}_j}}}{1-p}\)</span>.
In addition, the estimated standard errors, <span class="math inline">\(\widehat{\sigma_{X_{(1-\pi)p,j}}}\)</span> and
<span class="math inline">\(\widehat{\sigma_{X_{(1-\pi)(1-p),j}}}\)</span> can
be approximated by …, respectively.</p>
<p>Therefore, the values for each variant <span class="math inline">\(j=1, \dots, n_{\text{sig}}\)</span> in which all
genetic variants <span class="math inline">\(j=1, \dots,
n_{\text{sig}}\)</span> have been selected in the first <span class="math inline">\(\pi\)</span> fraction, i.e. satisfy the
condition…, are inputted into the MR method of choice.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_2.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_2.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
<p>In order to reduce the variability in the final MR-SimSS causal
effect estimate and ensure its stability, we suggest repeating this
simulated sample splitting process many times, at least <span class="math inline">\(N_\text{iter} = 1,000\)</span> iterations.</p>
<p>The final overall estimate for <span class="math inline">\(\hat\beta\)</span>, the exposure-outcome causal
effect, is then equivalent to the average of the estimates generated by
each iteration of the process.</p>
<p>An estimate of the causal effect obtained at iteration <span class="math inline">\(k\)</span> of the simulated sample splitting
process can be denoted by <span class="math inline">\(\hat\beta^{(k)}\)</span>. For example, we could
have <span class="math inline">\(\hat\beta^{(k)} =
\hat\beta^{(k)}_{\text{IVW}}\)</span> or <span class="math inline">\(\hat\beta^{(k)} =
\hat\beta^{(k)}_{\text{RAPS}}\)</span>, dependent on which summary-level
MR method is employed. The final estimate for the causal effect supplied
by MR-SimSS is then denoted by <span class="math inline">\(\overline{\hat\beta} = \frac{1}{N_{\text{iter}}}
\sum_{k=1}^{N_{\text{iter}}} \hat\beta^{(k)}\)</span>, the average of
the estimates obtained at each iteration.</p>
<p>A suitable value for <span class="math inline">\(\text{se}(\overline{\hat\beta})\)</span>, the
standard error of the MR-SimSS causal effect estimate, can be obtained
using the following expression:</p>
<p><span class="math display">\[\text{se}(\overline{\hat\beta}) =
\sqrt{\frac{\sum_{k=1}^{N_{\text{iter}}} \left[ \left( \text{se}\left(
\hat\beta^{(k)} \right)\right)^2 \right] - \sum_{k=1}^{N_{\text{iter}}}
\left[ \hat\beta^{(k)} -
\overline{\hat\beta}\right]^2}{N_\text{iter}}}\]</span> Note that <span class="math inline">\(\text{se}\left(\hat\beta^{(k)}\right)\)</span> is
the standard error of the causal effect estimate at iteration <span class="math inline">\(k\)</span>, which is typically provided by the
summary-level MR method of choice.</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Note:</strong> The
asymptotic conditional distribution equation suggests that in order to
implement MR-SimSS, if the number of overlapping samples involved in the
exposure and outcome GWASs, <span class="math inline">\(N_\text{overlap}\)</span>, is known to differ from
zero, <span class="math inline">\(N_\text{overlap} \neq 0\)</span>, then
this number <span class="math inline">\(N_\text{overlap}\)</span> as
well as the correlation between the exposure and outcome, <span class="math inline">\(\rho\)</span>, must be provided. In general, <span class="math inline">\(N_\text{overlap}\)</span> may be known, especially
if the GWASs have been performed on a single fully overlapping sample,
and an estimate for <span class="math inline">\(\rho\)</span> could be
obtained from the literature. However, if neither <span class="math inline">\(N_\text{overlap}\)</span> nor an estimate for
<span class="math inline">\(\rho\)</span> can be supplied, an
alternative strategy is provided <a href="">here</a> which seeks to
estimate a value for <span class="math inline">\(\lambda =
\frac{N_\text{overlap}\rho}{\sqrt{N_X N_Y}}\)</span>, using only the
available <strong>summary-level data of the genome-wide set of genetic
variants</strong> i.e. <span class="math inline">\(\{
\widehat{\gamma_j}, \widehat{\sigma_{X_j}}, \widehat{\Gamma_j},
\widehat{\sigma_{Y_j}}\}\)</span>.</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Note:</strong> It is
mentioned that several iterations of our approach, ideally <span class="math inline">\(N_{\text{iter}} \approx 1000\)</span> iterations,
is required in order to ensure a stable causal effect estimate.
Unfortunately, repeatedly simulating values for <span class="math inline">\(\{\gamma_{\pi_j}, \Gamma_{\pi_j}\}\)</span> and
<span class="math inline">\(\{\gamma_{{(1-\pi)p}_j},
\Gamma_{{(1-\pi)p}_j}\}\)</span> for SNPs <span class="math inline">\(j
= 1, \dots ,N\)</span> several times can make MR-SimSS a computationally
intensive process, especially in comparison with other summary-level MR
methods. Recognising that irrespective of the genetic architecture of
the exposure, there will likely be many variants which will never be
selected across the <span class="math inline">\(N_\text{iter}\)</span>
iterations, we therefore establish a subset of genetic variants
according to the cumulative sums of their probabilities, <span class="math inline">\(f\left( z_{X_{\pi,j}} &gt; 5.45\right)\)</span>.
Further details regarding can be found here.</p>
</div>
<div class="section level3">
<h3 id="assumptions-made-by-mr-simss">Assumptions made by MR-SimSS<a class="anchor" aria-label="anchor" href="#assumptions-made-by-mr-simss"></a>
</h3>
<p>As with many statistical methods, the MR-SimSS framework makes <span style="color:darkblue;"><strong>several assumptions</strong></span>:</p>
<ul>
<li>
Firstly, the genetic association estimates with respect to both exposure
and outcome are assumed to be <strong>normally distributed</strong> for
each <span class="math inline">\(j \in \{1, \dots, n\}\)</span>,
i.e. <span class="math inline">\(\widehat{\gamma_j} \sim
N\left(\gamma_j, \sigma_{X_j}^2 \right)\)</span> and <span class="math inline">\(\widehat{\Gamma_j} \sim N\left(\Gamma_j,
\sigma_{Y_j}^2 \right)\)</span>. This normality assumption frequently
appears in the MR literature and is considered to be reasonable, based
on asymptotic considerations, since <u>estimated associations are
obtained from large GWASs</u>.
</li>
<li>
It is assumed that the variance ratio <span class="math inline">\(\frac{\sigma_{X_j}^2}{\sigma_{Y_j}^2}\)</span>for
<span class="math inline">\(j=1, \dots, n\)</span> is both bounded away
from zero and infinity.
</li>
<li>
The set of random variables <strong><span class="math inline">\(\{\widehat{\gamma_j}, j=1, \dots, n\}\)</span> are
assumed to be mutually independent</strong>, while similarly,
<strong><span class="math inline">\(\{\widehat{\Gamma_j}, j=1, \dots, n
\}\)</span> are assumed to be mutually independent</strong>. However,
<span class="math inline">\(\gamma_j\)</span> and <span class="math inline">\(\Gamma_j\)</span> are not necessarily independent
due to participant overlap between the exposure and outcome data sets.
To satisfy this assumption regarding independence among variants, a
common step in MR studies is to first <u>employ LD pruning or clumping
to produce a large set of uncorrelated genetic variants</u>, which the
<span class="math inline">\(p\)</span> genetic instruments are then
selected from.
</li>
<li>
<p>The assumption that the <strong>standard deviations <span class="math inline">\(\sigma_{X_j}^2\)</span> and <span class="math inline">\(\sigma_{Y_j}^2\)</span> for <span class="math inline">\(j \in \{1, \dots, n\}\)</span> are known</strong>,
is also made here. In fact, most, if not all, MR analyses conducted
today assume that <span class="math inline">\(\widehat{\sigma_{X_j}}
\approx \sigma_{X_j}\)</span> and <span class="math inline">\(\widehat{\sigma_{Y_j}} \approx
\sigma_{Y_j}\)</span>. This practice is deemed appropriate by many as
large GWAS sample sizes ensure that the <u>variances of the genetic
association estimates are computed with a great degree of
accuracy</u>.</p>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="notation">Notation<a class="anchor" aria-label="anchor" href="#notation"></a>
</h3>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_MR/blob/main/table.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_MR/blob/main/table.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Amanda Forde.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
