<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MR-SimSS: The algorithm • mr.simss</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/lumen/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MR-SimSS: The algorithm">
<meta property="og:description" content="mr.simss">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mr.simss</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/index.html">Articles</a>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/amandaforde/mr.simss" class="external-link">Source</a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MR-SimSS: The algorithm</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/amandaforde/mr.simss/blob/HEAD/vignettes/MR-SimSS-algorithm.Rmd" class="external-link"><code>vignettes/MR-SimSS-algorithm.Rmd</code></a></small>
      <div class="hidden name"><code>MR-SimSS-algorithm.Rmd</code></div>

    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>A typical MR analysis is concerned with <strong>estimating the causal
effect, denoted by <span class="math inline">\(\beta\)</span>, of a
modifiable exposure, <span class="math inline">\(X\)</span> on a
health-related outcome, <span class="math inline">\(Y\)</span></strong>.
When variables which confound the exposure-outcome relationship have not
been observed, i.e. unmeasured confounders represented by <span class="math inline">\(U\)</span> are present, an MR study uses <span class="math inline">\(n\)</span> genetic variants, <span class="math inline">\(G_1, G_2, \dots, G_n\)</span>, in an attempt to
attain an unbiased causal effect estimate.</p>
<p>The image below displays a causal diagram representing the <u>
assumed relationship between genetic variant <span class="math inline">\(G_j\)</span>, exposure <span class="math inline">\(X\)</span> and outcome <span class="math inline">\(Y\)</span> in an MR analysis</u>. <span class="math inline">\(U\)</span> represents unobserved confounders,
while <span class="math inline">\(\gamma_j\)</span> is the effect of the
variant on the exposure and <span class="math inline">\(\beta\)</span>
is the true exposure-outcome causal effect. For a particular individual,
<span class="math inline">\(G_j\)</span> typically takes values 0, 1, or
2, <span class="math inline">\(G_j \in \{0,1,2\}\)</span>, depending on
the minor allele count or genotype of the individual at genetic variant
<span class="math inline">\(j\)</span>.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/causal_diagram.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/causal_diagram.jpg?raw=true" width="90%"></a>
</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Note:</strong> A
table is included at the bottom of the page (<a href="https://amandaforde.github.io/mr.simss/articles/MR-SimSS-algorithm.html#notation" class="external-link">here</a>)
detailing all notation used in this vignette.</p>
<p>Here, the focus is on <strong>summary-level MR</strong>, a form of MR
in which the analysis is conducted with summary statistics that have
been generated using two large non-overlapping, or partially
overlapping, exposure and outcome data sets. These summary statistics
typically take the form of variant-exposure association estimates,
variant-outcome association estimates and their corresponding standard
errors, i.e. <span class="math inline">\(\left( \widehat{\gamma_j},
\widehat{\sigma_{X_j}}\right)\)</span> and <span class="math inline">\(\left( \widehat{\Gamma_j},
\widehat{\sigma_{Y_j}}\right)\)</span> for each genetic variant <span class="math inline">\(j = 1, \dots, n\)</span>. To be more specific, the
exposure data is used to regress <span class="math inline">\(X\)</span>
on each <span class="math inline">\(G_j\)</span>, <span class="math inline">\(j = 1, \dots, n\)</span>, independently, yielding
regression coefficients, <span class="math inline">\(\widehat{\gamma_j}\)</span> and standard errors,
<span class="math inline">\(\widehat{\sigma_{X_j}}\)</span> in the
process. In a similar manner, the regression coefficients, <span class="math inline">\(\widehat{\Gamma_j}\)</span> and standard errors,
<span class="math inline">\(\widehat{\sigma_{Y_j}}\)</span> are
generated by regressing <span class="math inline">\(Y\)</span> on each
<span class="math inline">\(G_j\)</span> separately in the outcome data
set.</p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <span style="color:darkgreen;"><strong>Potential sample overlap is
permitted</strong></span> between the two data sets and therefore, it is
<u>not assumed that <span class="math inline">\(\widehat{\gamma_j}\)</span> and <span class="math inline">\(\widehat{\Gamma_j}\)</span> are independent</u>
for each genetic variant <span class="math inline">\(j\)</span>, i.e. it
is possible that <span class="math inline">\(\text{cov}\left(
\widehat{\gamma_j}, \widehat{\Gamma_j} \right) \neq 0\)</span>.</p>
</div>
<div class="section level3">
<h3 id="intuition-for-mr-simss-framework">Intuition for MR-SimSS framework<a class="anchor" aria-label="anchor" href="#intuition-for-mr-simss-framework"></a>
</h3>
<p>The intuition for MR-SimSS stems from a solution to <em>Winner’s
Curse</em> that could be implemented when individual-level data is
available.</p>
<div class="section level4">
<h4 id="individual-level">
<span style="color:darkblue;"> Individual-level: </span><a class="anchor" aria-label="anchor" href="#individual-level"></a>
</h4>
<p>In this setting, we first suppose that we have individual-level data,
i.e. genotype information, exposure measurements and/or outcome
measurements, for a total of <span class="math inline">\(N\)</span>
participants of the form <span class="math inline">\(\{\textbf{G},
\textbf{X}, \textbf{Y}\}\)</span>. Here, <span class="math inline">\(\textbf{G}\)</span> represents the matrix of
genotype information for each genetic variant and participant,
i.e. <span class="math inline">\(\textbf{G} = \{ \textbf{G}_1, \dots,
\textbf{G}_{N} \} \in \mathbb{R}^{N \times n}\)</span>, where <span class="math inline">\(\textbf{G}_{j} = \left[ G_{{1_j}},
\dots,G_{N_j}\right]^T\)</span> contains the genotype information for
the <span class="math inline">\(j\)</span>th variant, and <span class="math inline">\(\textbf{X} = \left[ X_1, \dots, X_{N}
\right]^T\)</span> is the vector of measured exposures. Similarly, <span class="math inline">\(\textbf{Y} = \left[ Y_1, \dots, Y_{N}
\right]^T\)</span> is the vector of outcome values. It is possible that
the vectors <span class="math inline">\(\mathbf{X}\)</span> and <span class="math inline">\(\mathbf{Y}\)</span> may have several missing
values, as there may only be exposure or outcome values, but not both,
available for certain individuals. We let <span class="math inline">\(N_\text{overlap}\)</span> denote the number of
overlapping individuals, those who have measured values for both outcome
and exposure. The proportion of sample overlap, <span class="math inline">\(\frac{N_\text{overlap}}{N}\)</span>, can range
from 0 to 1, i.e. no overlap to full overlap.</p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Eliminating</strong>
<strong><em>Winner’s Curse</em></strong>:</p>
<p>If this full data set <span class="math inline">\(\{\mathbf{G},\mathbf{X},\mathbf{Y}\}\)</span> were
available, in order to avoid <em>Winner’s Curse</em>, we could first
<strong>randomly split it into two subsets</strong>, with approximately
<span class="math inline">\(\pi N\)</span> and <span class="math inline">\((1−\pi)N\)</span> individuals in each set,
respectively. Following this splitting, an exposure GWAS could be
performed using the individual-level data in the first fraction or
subset with <span class="math inline">\(\pi N\)</span> individuals,
<span class="math inline">\(\{\mathbf{G}_\pi,\mathbf{X}_\pi,\mathbf{Y}_\pi
\}\)</span>, yielding <u> variant-exposure association estimates, <span class="math inline">\(\widehat{\gamma_{\pi_j}}\)</span>for each
variant</u> <span class="math inline">\(j = 1, \dots,n\)</span>. These
estimated associations could then be used for the <u>sole purpose of
selecting genetic instruments</u>. This would ensure that <em>Winner’s
Curse</em> is avoided as the other independent subset, <span class="math inline">\(\{\mathbf{G}_{1-\pi},\mathbf{X}_{1-\pi},\mathbf{Y}_{1-\pi}\}\)</span>,
would be used to perform an exposure and outcome GWAS, generating
variant-exposure, <span class="math inline">\(\widehat{\gamma_{{1-\pi}_j}}\)</span>, and
variant-outcome association estimates, <span class="math inline">\(\widehat{\Gamma_{{1-\pi}_j}}\)</span>, for the MR
study. This process is depicted in the figure below.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_1.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_1.jpg?raw=true" width="90%"></a>
</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Weak instrument
bias</strong>:</p>
<p>Recall from the first vignette, <a href="https://amandaforde.github.io/mr.simss/articles/wc_wib.html" class="external-link">“Winner’s
Curse and weak instrument bias in MR”</a>, that if the variant-exposure
and variant-outcome association estimates used in the MR analysis are
derived from a single sample or two largely overlapping samples and if
several weak instruments are incorporated, then bias towards the
confounded observational association will be introduced into the MR
causal effect estimate. When individual-level data is available, a
potential solution to this as well as <em>Winner’s Curse</em>, would be
to consider <strong>splitting the <span class="math inline">\(1-
\pi\)</span> fraction into two fractions, <span class="math inline">\(p\)</span> and <span class="math inline">\(1-p\)</span></strong>. As illustrated in the image
below, the dataset <span class="math inline">\(\{\mathbf{G}_{(1-\pi)(p)},\mathbf{X}_{(1-\pi)(p)},\mathbf{Y}_{(1-\pi)(p)}\}\)</span>
could be used to perform an exposure GWAS and supply variant-exposure
association estimates, <span class="math inline">\(\widehat{\gamma_{{(1-\pi)(p)}_j}}\)</span>, while
an outcome GWAS could be conducted with <span class="math inline">\(\{\mathbf{G}_{(1-\pi)(1-p)},\mathbf{X}_{(1-\pi)(1-p)},\mathbf{Y}_{(1-\pi)(1-p)}\}\)</span>,
giving variant-outcome association estimates, <span class="math inline">\(\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}\)</span>. A
suitable two-sample summary-level MR method could then be applied to the
resulting summary statistics: <span class="math inline">\(\{
\widehat{\gamma_{{(1-\pi)(p)}_j}}, \widehat{\Gamma_{{(1-\pi)(1-p)}_j}}
\}\)</span>.</p>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_2.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/individual_2.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
<p>Unfortunately, this splitting process, such as the one illustrated
above, results in a <u>great loss in sample size, and hence power</u>.
However, a reduction in the variability of the resulting MR causal
effect estimate can be acquired by <strong>repeating the random
splitting and performing GWASs in each subset many, many times</strong>,
and then, averaging over the causal effect estimates obtained on each
iteration.</p>
<p>However, this is <strong>not a very feasible solution</strong>, as in
addition to <u>requiring individual-level data</u>, a major limitation
of this approach is its <u>severe computational demands</u>, in terms of
both run time and memory usage.</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>MR-SimSS seeks to
imitate this repeated sample splitting process, using only GWAS summary
statistics.</strong></p>
</div>
</div>
<div class="section level3">
<h3 id="how-does-mr-simss-work">How does MR-SimSS work?<a class="anchor" aria-label="anchor" href="#how-does-mr-simss-work"></a>
</h3>
<p>As MR-SimSS works with <strong>summary-level data</strong>, it
assumes that an exposure and outcome GWAS have been performed, providing
variant-exposure and variant-outcome association estimates, <span class="math inline">\(\{\widehat{\gamma_{j}},
\widehat{\Gamma_j}\}\)</span>, in which there <u>may be a number of
overlapping individuals between the two GWASs</u>, <span class="math inline">\(N_\text{overlap}\)</span>.</p>
<ol>
<li>
<p>Constructed with the intention to imitate the above repeated sample
splitting, <strong>MR-SimSS first considers splitting the dataset into
two fractions</strong>, <span class="math inline">\(\pi\)</span> and
<span class="math inline">\(1-\pi\)</span>. <u>Values are simulated for
the association estimates in the first fraction</u>, <span class="math inline">\(\{\widehat{\gamma_{{\pi}_j}},
\widehat{\Gamma_{{\pi}_j}}\}\)</span>, conditional on those in the full
dataset, <span class="math inline">\(\{\widehat{\gamma_{j}},
\widehat{\Gamma_j}\}\)</span>, i.e. values are simulated for <span class="math inline">\(\{\widehat{\gamma_{{\pi}_j}},
\widehat{\Gamma_{{\pi}_j}}\}\)</span>, using the <span style="color:darkblue;"><strong>asymptotic conditional
distribution</strong></span> of <span class="math inline">\(\begin{bmatrix} \widehat{\gamma_{\pi}} \\
\widehat{\Gamma_{\pi}} \end{bmatrix} \Bigm| \begin{bmatrix}
\widehat{\gamma} \\ \widehat{\Gamma} \end{bmatrix}\)</span>. For genetic
variant <span class="math inline">\(j\)</span>, the asymptotic
conditional distribution has been derived as:</p>
<p><span class="math display">\[
\left( \begin{bmatrix} \widehat{\gamma_{\pi}} \\ \widehat{\Gamma_{\pi}}
\end{bmatrix} \Bigm| \begin{bmatrix} \widehat{\gamma} \\
\widehat{\Gamma} \end{bmatrix} \right) \sim N \left( \begin{bmatrix}
\widehat{\gamma} \\ \widehat{\Gamma} \end{bmatrix}, \left(
\frac{1-\pi}{\pi} \begin{bmatrix} \sigma^2_X &amp; \sigma_X \sigma_Y
\frac{N_\text{overlap} \rho}{N_X N_Y}\\ \sigma_X \sigma_Y
\frac{N_\text{overlap} \rho}{N_X N_Y} &amp; \sigma^2_Y\end{bmatrix}
\right)\right)
\]</span></p>
Here, <span class="math inline">\(\rho\)</span> represents the
correlation between the exposure and the outcome, <span class="math inline">\(\rho = \text{cor}(X,Y)\)</span>, which may differ
from <span class="math inline">\(\beta\)</span>, the true
exposure-outcome causal effect, due to confounding. A detailed outline
of the derivation of this asymptotic distribution can be found <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html" class="external-link">here</a>.
</li>
<li>
The <strong>estimated standard errors</strong> of the simulated
association estimates in the first fraction, <span class="math inline">\(\{
\widehat{\sigma_{X_{\pi,j}}},\widehat{\sigma_{Y_{\pi,j}}}\}\)</span> can
be obtained by approximating then with <span class="math inline">\(\sigma_{X_{\pi,j}} \sim \sqrt{\frac{1}{\pi}} \cdot
\widehat{\sigma_{X_{j}}}\)</span> and <span class="math inline">\(\sigma_{Y_{\pi,j}} \sim \sqrt{\frac{1}{\pi}} \cdot
\widehat{\sigma_{Y_{j}}}\)</span>, respectively.
</li>
<li>
The <strong>genetic instruments are selected</strong> according to their
<span class="math inline">\(z\)</span>-statistics, <span class="math inline">\(z_{X_{\pi,j}} =
\frac{\widehat{\gamma_{\pi_j}}}{\widehat{\sigma_{X_{\pi,j}}}}\)</span>
in fraction <span class="math inline">\(\pi\)</span>, i.e. genetic
variants are chosen to be incorporated into the MR analysis which have
corresponding <span class="math inline">\(p\)</span>-values less than
the genome-wide significance threshold of <span class="math inline">\(5
\times 10^{-8}\)</span>.
</li>
<li>
For each genetic variant, the <u>association estimates that would have
been obtained if GWASs were conducted in the second fraction</u>, <span class="math inline">\(1−\pi\)</span>, can be generated using <span class="math inline">\(\widehat{\gamma_j} \approx
\pi\widehat{\gamma_{\pi_j}} +
(1-\pi)\widehat{\gamma_{{(1-\pi)}_j}}\)</span> and <span class="math inline">\(\widehat{\Gamma_j} \approx
\pi\widehat{\Gamma_{\pi_j}} +
(1-\pi)\widehat{\Gamma_{{(1-\pi)}_j}}\)</span>. Note that these
relationships hold due to the <strong>asymptotic linearity of maximum
likelihood estimates</strong> but therefore, are only approximate to
order <span class="math inline">\(\frac{1}{N}\)</span>. The
corresponding standard errors in the <span class="math inline">\(1-\pi\)</span> fraction are similarly approximated
by <span class="math inline">\(\widehat{\sigma_{X_{1-\pi_j}}} \sim
\sqrt{\frac{1}{1-\pi}} \cdot \widehat{\sigma_{X_{j}}}\)</span> and <span class="math inline">\(\widehat{\sigma_{Y_{1-\pi_j}}} \sim
\sqrt{\frac{1}{1-\pi}} \cdot \widehat{\sigma_{Y_{j}}}\)</span>.
</li>
<li>
<p>Applying a <strong>summary-level MR method</strong> of choice, such
as the IVW method, using the summary statistics <span class="math inline">\(\{\widehat{\gamma_{{1-\pi}_j}},
\widehat{\sigma_{X_{1-\pi_j}}}, \widehat{\Gamma_{{1-\pi}_j}},
\widehat{\sigma_{Y_{1-\pi_j}}} \}\)</span> of selected genetic variants,
i.e. those that satisfy the condition <span class="math inline">\(\left|
z_{X_{\pi_j}} \right| &gt; \Phi^{-1}\left(1- \frac{5 \times
10^{-8}}{2}\right)\)</span>, describes the <span style="color:darkblue;"><strong>2-split version of
MR-SimSS</strong></span>. Steps 1-5 are illustrated in the figure
below.</p>
</li>
</ol>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_1.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_1.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
<p><u>However</u>, as mentioned above, if there exists a large degree of
overlap between the sample sets used in the exposure and outcome GWASs,
then using association estimates derived from the same fraction is
likely to induce a notable amount of <strong>weak instrument
bias</strong> in the MR estimate <strong>in the direction of the
confounded observational exposure-outcome association</strong>.</p>
<p>Therefore, a <span style="color:darkblue;"><strong>3-split version of
MR-SimSS</strong></span> was proposed that would eliminate <em>Winner’s
Curse</em> bias, and <u>only result in weak instrument bias towards the
null, irrespective of the proportion of sample overlap</u>.</p>
<ol start="5">
<li>
After carrying out steps 1-4 above, the 3-split version of MR-SimSS
imitates the <strong>splitting of individual-level data in the <span class="math inline">\(1-\pi\)</span> fraction</strong> into two
fractions, <span class="math inline">\(p\)</span> and <span class="math inline">\(1− p\)</span>. Values are simulated for the
estimated variant-exposure and variant-outcome associations, <span class="math inline">\(\{\widehat{\gamma_{{(1-\pi)p}_j}},
\widehat{\Gamma_{{(1-\pi)p}_j}}\}\)</span>, respectively, conditional on
the generated association estimates in the <span class="math inline">\(1-\pi\)</span> fraction, <span class="math inline">\(\{\widehat{\gamma_{{(1-\pi)}_j}},
\widehat{\Gamma_{{(1-\pi)}_j}}\}\)</span>. The <strong>asymptotic
conditional distribution</strong> of <span class="math inline">\(\begin{bmatrix} \widehat{\gamma_{(1-\pi)p}} \\
\widehat{\Gamma_{(1-\pi)p}} \end{bmatrix} \Bigm| \begin{bmatrix}
\widehat{\gamma_{1-\pi}} \\ \widehat{\Gamma_{1-\pi}}
\end{bmatrix}\)</span> takes a similar format to the conditional
distribution included above - an expression for it can be found <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html" class="external-link">here</a>.
</li>
<li>
Using the <strong>asymptotic linearity of maximum likelihood
estimates</strong>, <span class="math inline">\(\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}\)</span> is
obtained using <span class="math inline">\(\widehat{\Gamma_{{(1-\pi)}_j}} \approx
p\widehat{\Gamma_{{(1-\pi)p}_j}}-(1-p)\widehat{\Gamma_{{(1-\pi)(1-p)}_j}}\)</span>,
with the estimated standard errors <span class="math inline">\(\widehat{\sigma_{X_{(1-\pi)p_j}}}\)</span> and
<span class="math inline">\(\widehat{\sigma_{X_{(1-\pi)(1-p)_j}}}\)</span>
again approximated in a similar manner to previously.
</li>
<li>
A <strong>summary-level MR method</strong> of choice, such as the IVW or
MR-RAPS methods is applied to the summary statistics <span class="math inline">\(\{\widehat{\gamma_{{(1-\pi)}p_j}},
\widehat{\sigma_{X_{(1-\pi)p_j}}}, \widehat{\Gamma_{{(1-\pi)(1-p)}_j}},
\widehat{\sigma_{Y_{(1-\pi)(1-p)_j}}} \}\)</span> of genetic variants
<span class="math inline">\(j=1, \dots, n_{\text{sig}}\)</span> that
have been selected in the first <span class="math inline">\(\pi\)</span>
fraction, i.e. those that satisfy the condition <span class="math inline">\(\left| z_{X_{\pi_j}} \right| &gt;
\Phi^{-1}\left(1- \frac{5 \times 10^{-8}}{2}\right)\)</span>. This
process is depicted in the figure below.
</li>
</ol>
<p><br></p>
<p align="center">
<a href="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_2.jpg?raw=true" class="external-link">
<img src="https://github.com/amandaforde/winnerscurse_sims/blob/main/summary_2.jpg?raw=true" width="100%"></a>
</p>
<p><br></p>
<ol start="8">
<li>
In order to <u>reduce the variability in the final MR-SimSS causal
effect estimate and ensure its stability</u>, we suggest
<strong>repeating this simulated sample splitting process many
times</strong>, at least <span class="math inline">\(N_\text{iter} =
1,000\)</span> iterations.
</li>
<li>
The <strong>final value for <span class="math inline">\(\hat\beta\)</span></strong>, the estimated
exposure-outcome causal effect, supplied by MR-SimSS is equivalent to
the <strong>average of the estimates generated by each
iteration</strong> of the process, i.e. <span class="math inline">\(\overline{\hat\beta} = \frac{1}{N_{\text{iter}}}
\sum_{k=1}^{N_{\text{iter}}} \hat\beta^{(k)}\)</span>. Here, the
estimate of the causal effect obtained at iteration <span class="math inline">\(k\)</span> of the simulated sample splitting
process is denoted by <span class="math inline">\(\hat\beta^{(k)}\)</span>.
</li>
<li>
<p>A suitable value for <span class="math inline">\(\text{se}(\overline{\hat\beta})\)</span>, the
<strong>standard error of the MR-SimSS causal effect estimate</strong>,
can be obtained using the following expression:</p>
<span class="math display">\[\text{se}(\overline{\hat\beta}) =
\sqrt{\frac{\sum_{k=1}^{N_{\text{iter}}} \left[ \left( \text{se}\left(
\hat\beta^{(k)} \right)\right)^2 \right] - \sum_{k=1}^{N_{\text{iter}}}
\left[ \hat\beta^{(k)} -
\overline{\hat\beta}\right]^2}{N_\text{iter}}}\]</span> Note that <span class="math inline">\(\text{se}\left(\hat\beta^{(k)}\right)\)</span> is
the standard error of the causal effect estimate at iteration <span class="math inline">\(k\)</span>, which is typically provided by the
summary-level MR method of choice. More details regarding the expression
for <span class="math inline">\(\text{se}(\overline{\hat\beta})\)</span>
can be found <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html" class="external-link">here</a>.
</li>
</ol>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Note:</strong> The
asymptotic conditional distribution equation, given in Step 1, suggests
that in order to implement MR-SimSS, if the number of overlapping
samples involved in the exposure and outcome GWASs, <span class="math inline">\(N_\text{overlap}\)</span>, is known to differ from
zero, <span class="math inline">\(N_\text{overlap} \neq 0\)</span>, then
this number <span class="math inline">\(N_\text{overlap}\)</span> as
well as the correlation between the exposure and outcome, <span class="math inline">\(\rho\)</span>, must be provided. In general, <span class="math inline">\(N_\text{overlap}\)</span> may be known, especially
if the GWASs have been performed on a single fully overlapping sample,
and an estimate for <span class="math inline">\(\rho\)</span> could be
obtained from the literature. However, especially when only GWAS summary
statistics are available, it can often be difficult to know the value of
<span class="math inline">\(N_\text{overlap}\)</span> or <span class="math inline">\(\rho\)</span>. Therefore, within its
functionality, MR-SimSS also has an <strong>alternative strategy which
seeks to estimate a value for <span class="math inline">\(\lambda =
\frac{N_\text{overlap}\rho}{\sqrt{N_X N_Y}}\)</span></strong>, using
only the available <strong>summary-level data of the <u>genome-wide
set</u> of genetic variants</strong> i.e. <span class="math inline">\(\{
\widehat{\gamma_j}, \widehat{\sigma_{X_j}}, \widehat{\Gamma_j},
\widehat{\sigma_{Y_j}}\}\)</span>. Additional details regarding this
approach, which is based on conditional likelihood, can be found <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html#estimating-the-correlation" class="external-link">here</a>.</p>
<p><br></p>
<p><span style="color: darkblue;"><span class="math inline">\(\star\)</span></span> <strong>Note:</strong>
Unfortunately, repeatedly simulating values for <span class="math inline">\(\{\gamma_{\pi_j}, \Gamma_{\pi_j}\}\)</span> and
<span class="math inline">\(\{\gamma_{{(1-\pi)p}_j},
\Gamma_{{(1-\pi)p}_j}\}\)</span> for thousands of genetic variants can
make MR-SimSS a very computationally intensive process, especially in
comparison with other summary-level MR methods. By recognising that
irrespective of the genetic architecture of the exposure, there will
likely be many variants which will never be selected across the <span class="math inline">\(N_\text{iter}\)</span> iterations, MR-SimSS
contains a <strong>computationally efficient feature</strong> which
first establishes a <strong>smaller subset of genetic variants</strong>
that will then be entered into the main algorithm. This subset of
genetic variants is constructed by first ordering all variants according
to their probabilities, <span class="math inline">\(f\left(
z_{X_{\pi,j}} &gt; 5.45\right)\)</span>, from smallest to largest and
then, obtaining the <u>cumulative sum of these probabilities</u> for
each variant. All SNPs with a cumulative probability greater than 0.05
are included in the subset. This process ensures that for a single
iteration of MR-SimSS, the <u>number of genetic instruments if the full
set of SNPs is used and the number of instruments if merely the subset
is used will be equal with probability at least 95%</u>. Further details
regarding can be found <a href="https://amandaforde.github.io/mr.simss/articles/derive-MR-SimSS.html#reducing-the-set-of-variants" class="external-link">here</a>.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="assumptions-made-by-mr-simss">Assumptions made by MR-SimSS<a class="anchor" aria-label="anchor" href="#assumptions-made-by-mr-simss"></a>
</h3>
<p>As with many statistical methods, the MR-SimSS framework makes <span style="color:darkblue;"><strong>several assumptions</strong></span>:</p>
<ul>
<li>
Firstly, the genetic association estimates with respect to both exposure
and outcome are assumed to be <strong>normally distributed</strong> for
each <span class="math inline">\(j \in \{1, \dots, n\}\)</span>,
i.e. <span class="math inline">\(\widehat{\gamma_j} \sim
N\left(\gamma_j, \sigma_{X_j}^2 \right)\)</span> and <span class="math inline">\(\widehat{\Gamma_j} \sim N\left(\Gamma_j,
\sigma_{Y_j}^2 \right)\)</span>. This normality assumption frequently
appears in the MR literature and is considered to be reasonable, based
on asymptotic considerations, since <u>estimated associations are
obtained from large GWASs</u>.
</li>
<li>
It is assumed that the variance ratio <span class="math inline">\(\frac{\sigma_{X_j}^2}{\sigma_{Y_j}^2}\)</span>for
<span class="math inline">\(j=1, \dots, n\)</span> is both bounded away
from zero and infinity.
</li>
<li>
The set of random variables <strong><span class="math inline">\(\{\widehat{\gamma_j}, j=1, \dots, n\}\)</span> are
assumed to be mutually independent</strong>, while similarly,
<strong><span class="math inline">\(\{\widehat{\Gamma_j}, j=1, \dots, n
\}\)</span> are assumed to be mutually independent</strong>. However,
<span class="math inline">\(\gamma_j\)</span> and <span class="math inline">\(\Gamma_j\)</span> are not necessarily independent
due to participant overlap between the exposure and outcome data sets.
To satisfy this assumption regarding independence among variants, a
common step in MR studies is to first <u>employ LD pruning or clumping
to produce a large set of uncorrelated genetic variants</u>, which the
<span class="math inline">\(p\)</span> genetic instruments are then
selected from.
</li>
<li>
<p>The assumption that the <strong>standard deviations <span class="math inline">\(\sigma_{X_j}^2\)</span> and <span class="math inline">\(\sigma_{Y_j}^2\)</span> for <span class="math inline">\(j \in \{1, \dots, n\}\)</span> are known</strong>,
is also made here. In fact, most, if not all, MR analyses conducted
today assume that <span class="math inline">\(\widehat{\sigma_{X_j}}
\approx \sigma_{X_j}\)</span> and <span class="math inline">\(\widehat{\sigma_{Y_j}} \approx
\sigma_{Y_j}\)</span>. This practice is deemed appropriate by many as
large GWAS sample sizes ensure that the <u>variances of the genetic
association estimates are computed with a great degree of
accuracy</u>.</p>
</li>
</ul>
<style type="text/css">
   table,
    th,
    td {
         border: 1px solid black;
  font-size: 12.5px; 
    }
 
    th,
    td {
        padding: 8px;
    }
    td:nth-child(odd){
    text-align: center;
}
    th {
        background-color: #38d1a4;
    }
</style>
<p><br></p>
</div>
<div class="section level3">
<h3 id="notation">Notation<a class="anchor" aria-label="anchor" href="#notation"></a>
</h3>
<table style="width:95%" class="table">
<tr>
<th>
<strong>Symbol</strong>
</th>
<th>
<strong>What does it represent?</strong>
</th>
</tr>
<tr>
<td>
<span class="math inline">\(\beta\)</span>
</td>
<td>
True causal effect of modifiable exposure on health-related outcome
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\hat\beta\)</span>
</td>
<td>
<u>Estimated</u> exposure-outcome causal effect
</td>
</tr>
<tr>
<td>
<span class="math inline">\(X\)</span>
</td>
<td>
Exposure
</td>
</tr>
<tr>
<td>
<span class="math inline">\(Y\)</span>
</td>
<td>
Outcome
</td>
</tr>
<tr>
<td>
<span class="math inline">\(U\)</span>
</td>
<td>
Unmeasured variables which confound the exposure-outcome relationship
</td>
</tr>
<tr>
<td>
<span class="math inline">\(G_j\)</span>
</td>
<td>
Genotype of genetic variant <span class="math inline">\(j\)</span>,
<span class="math inline">\(G_j \in \{0,1,2\}\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\gamma_j\)</span>
</td>
<td>
True effect of genetic variant <span class="math inline">\(j\)</span> on
the exposure
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\hat\gamma_j\)</span>
</td>
<td>
<u>Estimated</u> association between genetic variant <span class="math inline">\(j\)</span> and the exposure
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\gamma_{*_j}}\)</span>
</td>
<td>
<u>Estimated</u> association between genetic variant <span class="math inline">\(j\)</span> and the exposure in fraction <span class="math inline">\(*\)</span> of the full dataset
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\sigma_{X_j}\)</span>
</td>
<td>
True standard error of the variant-exposure association estimate, <span class="math inline">\(\widehat{\gamma_j}\)</span>, for variant <span class="math inline">\(j\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\sigma_{X_j}}\)</span>
</td>
<td>
<u>Estimated</u> standard error of the variant-exposure association
estimate, <span class="math inline">\(\widehat{\gamma_j}\)</span>, for
variant <span class="math inline">\(j\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\sigma_{X_{*,j}}}\)</span>
</td>
<td>
<u>Estimated</u> standard error of the variant-exposure association
estimate, <span class="math inline">\(\widehat{\gamma_{*_j}}\)</span>,
for variant <span class="math inline">\(j\)</span> in fraction <span class="math inline">\(*\)</span> of the full dataset
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\Gamma_j}\)</span>
</td>
<td>
<u>Estimated</u> association between genetic variant <span class="math inline">\(j\)</span> and the outcome
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\Gamma_{*_j}}\)</span>
</td>
<td>
<u>Estimated</u> association between genetic variant <span class="math inline">\(j\)</span> and the outcome in fraction <span class="math inline">\(*\)</span> of the full dataset
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\sigma_{Y_j}\)</span>
</td>
<td>
True standard error of the variant-outcome association estimate, <span class="math inline">\(\widehat{\Gamma_j}\)</span>,for variant <span class="math inline">\(j\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\sigma_{Y_j}}\)</span>
</td>
<td>
<u>Estimated</u> standard error of the variant-outcome association
estimate, <span class="math inline">\(\widehat{\Gamma_j}\)</span>,for
variant <span class="math inline">\(j\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\widehat{\sigma_{Y_{*,j}}}\)</span>
</td>
<td>
<u>Estimated</u> standard error of the variant-outcome association
estimate, <span class="math inline">\(\widehat{\Gamma_j}\)</span>,for
variant <span class="math inline">\(j\)</span> in fraction <span class="math inline">\(*\)</span> of the full dataset
</td>
</tr>
<tr>
<td>
<span class="math inline">\(n\)</span>
</td>
<td>
Number of measured genetic variants
</td>
</tr>
<tr>
<td>
<span class="math inline">\(N\)</span>
</td>
<td>
Number of participants in the full data set, i.e. number of individuals
with genotype information, exposure and/or outcome measurements
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\{\mathbf{G},
\mathbf{X},\mathbf{Y}\}\)</span>
</td>
<td>
Full individual-level dataset in which <span class="math inline">\(\mathbf{G}\)</span> represents the matrix of
genotype information for each genetic variant and participant,
i.e. <span class="math inline">\(\mathbf{G} = \{\mathbf{G}_1, \dots,
\mathbf{G}_N\}\in \mathbb{R}^{N\times n}\)</span>, where <span class="math inline">\(\mathbf{G}_j = [G_{1j}, \dots, G_{Nj}]^T\)</span>
contains the genotype information for the <span class="math inline">\(j\)</span>th variant, <span class="math inline">\(\mathbf{X} = [X_1, \dots,X_N]^T\)</span> is the
vector of measured exposures, and <span class="math inline">\(\mathbf{Y}
= [Y_1, \dots, Y_N]^T\)</span> is the vector of outcome values
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\{\mathbf{G}_{*},
\mathbf{X}_{*},\mathbf{Y}_{*}\}\)</span>
</td>
<td>
Individual-level data set, similar to that above, but only containing
data for those individuals in fraction <span class="math inline">\(*\)</span> of the full dataset
</td>
</tr>
<tr>
<td>
<span class="math inline">\(N_\text{overlap}\)</span>
</td>
<td>
Number of overlapping individuals, i.e. number of participants who have
measured values for both outcome and exposure
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\pi\)</span>
</td>
<td>
First fraction in which the full dataset is split into – fraction <span class="math inline">\(\pi\)</span> is used to select genetic instruments
</td>
</tr>
<tr>
<td>
<span class="math inline">\(p\)</span>
</td>
<td>
Second fraction in which the full dataset is split into – fraction <span class="math inline">\(1-\pi\)</span> is split into fractions <span class="math inline">\(p\)</span> and <span class="math inline">\(1-p\)</span> to estimate variant-exposure and
variant-outcome associations
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\rho\)</span>
</td>
<td>
Correlation between the exposure and the outcome, i.e. <span class="math inline">\(\rho = \text{cor}(X,Y)\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(z_{X_{\pi,j}}\)</span>
</td>
<td>
<span class="math inline">\(z\)</span>-statistic of variant <span class="math inline">\(j\)</span> in fraction <span class="math inline">\(\pi\)</span> of the dataset, i.e. <span class="math inline">\(z_{X_{\pi,j}} =
\frac{\widehat{\gamma_{\pi_j}}}{\widehat{\sigma_{X_{\pi,j}}}}\)</span> -
used to select genetic instruments
</td>
</tr>
<tr>
<td>
<span class="math inline">\(n_\text{sig}\)</span>
</td>
<td>
Number of genetic instruments which have been selected using the
(simulated) summary statistics from the first fraction, <span class="math inline">\(\pi\)</span>, of the dataset
</td>
</tr>
<tr>
<td>
<span class="math inline">\(N_\text{iter}\)</span>
</td>
<td>
Number of iterations of sample splitting performed by MR-SimSS
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\hat\beta^{(k)}\)</span>
</td>
<td>
Causal effect estimate produced by MR-SimSS on iteration <span class="math inline">\(k\)</span>
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\overline{\hat\beta}\)</span>
</td>
<td>
Final estimate for the exposure-outcome causal effect supplied by
MR-SimSS, i.e. average of the estimates generated at each iteration
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\text{se}(\overline{\hat\beta})\)</span>
</td>
<td>
Standard error of MR-SimSS causal effect estimate
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\text{se}\left(\hat\beta^{(k)}\right)\)</span>
</td>
<td>
Standard error of causal effect estimate at iteration <span class="math inline">\(k\)</span>, generally supplied by chosen
summary-level MR method
</td>
</tr>
<tr>
<td>
<span class="math inline">\(\lambda\)</span>
</td>
<td>
Symbol representing <span class="math inline">\(\frac{N_{\text{overlap}}\rho}{\sqrt{N_X
N_Y}}\)</span> in which <span class="math inline">\(N_X\)</span> is the
number of individuals in the full dataset with measured exposures,
i.e. exposure GWAS sample size, and <span class="math inline">\(N_Y\)</span> is the number of individuals in the
full dataset with measured outcomes, i.e. outcome GWAS sample size
</td>
</tr>
</table>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Amanda Forde.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
